format/IPC.md

```diff
@@ -67,15 +67,18 @@ We provide a streaming format for record batches. It is presented as a sequence
 of encapsulated messages, each of which follows the format above. The schema
 comes first in the stream, and it is the same for all of the record batches
 that follow. If any fields in the schema are dictionary-encoded, one or more
-`DictionaryBatch` messages will follow the schema.
+`DictionaryBatch` messages will be included. `DictionaryBatch` and
+`RecordBatch` messages may be interleaved, but before any dictionary key is used
+in a `RecordBatch` it should be defined in a `DictionaryBatch`.
 
 ```
 <SCHEMA>
 <DICTIONARY 0>
 ...
-<DICTIONARY k - 1>
 <RECORD BATCH 0>
 ...
+<DICTIONARY k - 1>
+...
```

Comment:
This is slightly misleading. I believe that all dictionaries need to appear before the first record batch, but then arbitrary delta batches may appear after the first record batch

Reply:
Yeah thats fair, I can tweak it to make it clear that any dictionaries after the first record batch should be modifying the originals.